broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java

```diff
@@ -31,6 +31,8 @@
    */
   ZeebeDb openDb();
 
+  void closeDb() throws Exception;
```

Comment:
:question:  What is the difference to the `close()` method? The state controller already implements `AutoCloseable`.

Reply:
DB is opened by ZeebeDBPartitionStep. So it should be closed by the same. In future StateControllerStep will be only installed once at the startup, and closed only when the broker is closed. But we have to close/re-open zeebedb during role transitions.

Reply:
Do you mean that `StateControllerImpl#closeDb()` and `StateControllerImpl#close()`will behave differently (i.e. it doesn't remove the runtime forlder)? 
Currently, `close()` delegates to `closeDb()`.

Reply:
Both behave similar now. May be we don't need to close db in ``StateControllerImpl#close()`. We can assume that ZeebeDBStep will close it and remove the runtime folder.

Reply:
I'm okay if you see a need in it :+1: